File: api/src/main/java/akkamon/api/models/EventType.java
package akkamon.api.models;

import com.google.gson.annotations.SerializedName;

public enum EventType {
    @SerializedName("TrainerRegistrationRequestEvent")
    TRAINER_REGISTRATION_REQUEST,

    @SerializedName("TrainerRegistrationReplyEvent")
    TRAINER_REGISTRATION_REPLY,

    @SerializedName("HeartBeat")
    HEART_BEAT,

    @SerializedName("StartMoving")
    START_MOVING,

    @SerializedName("NewTilePos")
    NEW_TILE_POS,

    @SerializedName("StopMoving")
    STOP_MOVING,

    @SerializedName("SocketClosed")
    SOCKET_CLOSED,

    @SerializedName("InteractionRequestEvent")
    INTERACTION_REQUEST

}

File: client/src/akkamon/client/EventType.ts
export interface AkkamonEvent {
    type: EventType
}

export enum EventType {
    HEART_BEAT = "HeartBeat",
    TRAINER_REGISTRATION_REQUEST = "TrainerRegistrationRequestEvent",
    TRAINER_REGISTRATION_REPLY = "TrainerRegistrationReplyEvent",
    START_MOVING =  "StartMoving",
    STOP_MOVING = "StopMoving",
    NEW_TILE_POS = "NewTilePos",
    INTERACTION_REQUEST = "InteractionRequestEvent",
    INTERACTION_REPLY = "InteractionReplyEvent",
    INTERACTION_ABORTED = "InteractionAbortedEvent",
    INTERACTION_START = "InteractionStarting"
}

